test/unit/routesTest.js

```diff
@@ -348,30 +374,40 @@ describe('Roosevelt routes Section Test', function () {
-      let bool503 = false
-      for (let x = 0; x < 10; x++) {
-        request('http://localhost:43711')
-        .get('/HTMLTest')
-        .expect(200, (err, res) => {
-          if (err && res !== undefined && bool503 === false) {
-            assert.equal(res.status, 503)
-            bool503 = true
-            // sample the response text
-            const test1 = res.text.includes('503 Service Unavailable')
-            const test2 = res.text.includes('The requested URL /HTMLTest is temporarily unavailable at this time')
-            const test3 = res.text.includes('localhost:43711')
-            // check to make sure that all specific pharses are there
-            assert.equal(test1, true)
-            assert.equal(test2, true)
-            assert.equal(test3, true)
-            testApp.kill()
-            done()
-          }
-        })
-      }
+      async.series([
+        function (cb) { request('http://localhost:43711').get('/HTMLTest').expect(200, (err, res) => { if (err) { cb(res, null) } else { cb(null, res) } }) },
+        function (cb) { request('http://localhost:43711').get('/HTMLTest').expect(200, (err, res) => { if (err) { cb(res, null) } else { cb(null, res) } }) },
+        function (cb) { request('http://localhost:43711').get('/HTMLTest').expect(200, (err, res) => { if (err) { cb(res, null) } else { cb(null, res) } }) },
+        function (cb) { request('http://localhost:43711').get('/HTMLTest').expect(200, (err, res) => { if (err) { cb(res, null) } else { cb(null, res) } }) },
+        function (cb) { request('http://localhost:43711').get('/HTMLTest').expect(200, (err, res) => { if (err) { cb(res, null) } else { cb(null, res) } }) },
+        function (cb) { request('http://localhost:43711').get('/HTMLTest').expect(200, (err, res) => { if (err) { cb(res, null) } else { cb(null, res) } }) },
+        function (cb) { request('http://localhost:43711').get('/HTMLTest').expect(200, (err, res) => { if (err) { cb(res, null) } else { cb(null, res) } }) },
+        function (cb) { request('http://localhost:43711').get('/HTMLTest').expect(200, (err, res) => { if (err) { cb(res, null) } else { cb(null, res) } }) },
+        function (cb) { request('http://localhost:43711').get('/HTMLTest').expect(200, (err, res) => { if (err) { cb(res, null) } else { cb(null, res) } }) },
+        function (cb) { request('http://localhost:43711').get('/HTMLTest').expect(200, (err, res) => { if (err) { cb(res, null) } else { cb(null, res) } }) }
+      ], (err, result) => {
+        // if any one of the test brings an error. (one of them should). Check to see if the err response text has the custom 503 page we put in there
+        if (err) {
+          // sample the response text
+          const test1 = err.res.text.includes('503 Service Unavailable')
+          const test2 = err.res.text.includes('The requested URL /HTMLTest is temporarily unavailable at this time')
+          const test3 = err.res.text.includes('localhost:43711')
```

Comment:
probably don't want to hardcode the port in here in case the default port ever changes